WordPress/openverse-frontend
# Convert the search store to Pinia

## Fixes


Fixes #1037 by @obulat
Fixes #1121 by @sarayourfriend
Fixes #1128 by @dhruvkb 

## Description


This PR converts the search store from Vuex to Pinia and uses TypeScript.

For the search store state, we have 3 main pieces of data:
- search term (`q` parameter in the API request query)
- search type (is converted to path segment after `/search/`), and used throughout the codebase
- filters - these are lists of objects in each filter category (in the filter sidebar / filter modal) that have the code of the filter, the name (used to get the i18n string for it) and `checked` parameter. The set of filters that is displayed is dependent on the currently selected search type.

There is also a `query` object that holds the data necessary to set the URL for the API request. This was a separate object that was updated manually before. But in light of #1121, this PR converted it into a computed.
The search store uses the filters object as the single source of truth, and uses that to compute the `searchQueryParams` to be used for the API request.

There are currently 37 files that changed, but most of the changes in the components are renaming. The most significant changes are in the `stores/search.ts` and `VHeader.vue` and `search-query-transform.ts` files.

## Testing Instructions

Try out running the app and:
- searching for different words using the homepage and the searchbar in the header
- try checking and unchecking filters
- changing the search type using both the 'Content link' and the search type switcher at the top of the page
- clearing the filters using 'Clear filters' button.
- going to the detail page and back to the search page.
- going to the detail page and searching from it (it should go to the All content view with no filters applied)

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
